适配LiquidGlass、UIScene、修复一系列issue

CloudlessMoon opened the pull request:
适配LiquidGlass、UIScene、修复一系列issue

CloudlessMoon committed:
1、修复一系列issue
2、ruanmei-fix，适配LiquidGlass

CloudlessMoon committed:
UIBarItem (QMUIBadge)，适配LiquidGlass

qmui_updatesIndicatorView、qmui_badgeView需要改成block实现了，返回一个View实例

CloudlessMoon committed:
移除不需要的NSAssert

CloudlessMoon committed:
QMUIPopupContainerView，增加一个Assert

CloudlessMoon committed:
1、QMUIKeyboardManager 适配iOS 26
2、QMUIModalPresentationViewController 兼容键盘已经弹起的情况

CloudlessMoon committed:
https://github.com/Tencent/QMUI_iOS/issues/1634，修复业务设置UIWindow overrideUserInterfaceStyle 为light或者dark后，无法触发回调的问题

CloudlessMoon committed:
https://github.com/Tencent/QMUI_iOS/issues/1668，修复qmuisl_stepControls 设置userInteractionEnabled错误的问题

CloudlessMoon committed:
https://github.com/Tencent/QMUI_iOS/issues/1668，修复qmuisl_stepControls 设置userInteractionEnabled错误的问题

CloudlessMoon committed:
iOS 26，修复转场动画的_QMUITransitionNavigationBar y值错误的问题

CloudlessMoon committed:
iOS 26，_QMUITransitionNavigationBar 设置frame 后，立即触发layoutIfNeeded，否则在手势返回时会导致有个跟随的动画

CloudlessMoon committed:
QMUITheme, [UITraitCollection currentTraitCollection] 改为 UIScreen.mainScreen.traitCollection更准确些，[UITraitCollection currentTraitCollection]是基于当前上下文

CloudlessMoon committed:
QMUIModalPresentationViewController、QMUIPopupContainerView增加- (void)showInWindow:(nullable UIWindow *)window，用于多个windowScene的情况下， 业务可以选择某个window去show，避免被遮挡

CloudlessMoon committed:
UIWindow隐藏时，将windowScene置空

CloudlessMoon committed:
QMUIModalPresentationViewController适配键盘已经弹起的情况，随后又落下的情况

CloudlessMoon committed:
QMUIPopupContainerView，当上下或者左右没有足够的空间时，不能直接改变maximumHeight或者maximumWidth，不然当有足够的空间时，会因maximumHeight/maximumWidth计算错误

CloudlessMoon committed:
QMUITheme, 使用UIScreen的_setDefaultTraitCollection更准确

CloudlessMoon committed:
qmui_addUserInterfaceStyleWillChangeObserver，mac上使用UIScreen的_setDefaultTraitCollection，其他平台使用_parentWillTransitionToTraitCollection:

zhenian commented:
> [![CLA assistant check](https://camo.githubusercontent.com/33367afe38a8271d65c4a44ccbeac76fee99d00769e86aeb0712cd8bfbd725bf/68747470733a2f2f636c612e6f70656e736f757263652e74656e63656e742e636f6d2f70756c6c2f62616467652f6e6f745f7369676e6564)](https://cla.opensource.tencent.com/Tencent/QMUI_iOS?pullRequest=1672) Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/QMUI_iOS?pullRequest=1672) before we can accept your contribution.You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/QMUI_iOS?pullRequest=1672) it.

看流程检查， 需要 @CloudlessMoon   ，签署一个同意协议 Contributor License Agreement 才能有机会合代码接受贡献

CloudlessMoon committed:
https://github.com/Tencent/QMUI_iOS/issues/1680

CloudlessMoon committed:
Revert "https://github.com/Tencent/QMUI_iOS/issues/1680"

This reverts commit ebb9d3a668442d867531ab6672861e918f54c65f.

CloudlessMoon committed:
暂时性修复https://github.com/Tencent/QMUI_iOS/issues/1680，此为UIKit的bug

CloudlessMoon committed:
优化获取hx_keyWindow的逻辑，当connectedScenes里找不到keyWindow时，使用UIApplication.shared.keyWindow去获取

CloudlessMoon committed:
iOS 26液态玻璃下，转场动画可以被打断，不需要QMUILogWarn了

CloudlessMoon committed:
QMUITextField, clearButton也应用上QMUITextField.textInsets

CloudlessMoon committed:
1、UIApplication，qmui_keyWindow，iOS 15.0 以上使用 windowScene.keyWindow
2、[ qmui_getBoundBOOLForKey:@"QMUIAddedObserver"] 使用qmui_addedObserver

CloudlessMoon committed:
修复切换主题后某些UI未及时响应的问题

CloudlessMoon committed:
修复切换主题后某些UI未及时响应的问题

CloudlessMoon committed:
① 修复qmui_topViewController失效的问题，qmui_endedTransitionTopViewController需要用新的私有方法
② 修复- (void) handleKeyWithString:(id)arg1 forKeyEvent:(id)arg2 executionContext:(id)arg3; 失效的问题

nouslove commented:
shouldPopViewControllerByBackButtonOrPopGesture:无法拦截手势或者按钮事件

CloudlessMoon committed:
① iOS 26，返回按钮拦截失效
② _UINavigationBarContentView更换为UIKit.NavigationBarContentView

CloudlessMoon committed:
Debug下hook [UINavigationBar setStandardAppearance:]，避免无意义的耗时

CloudlessMoon committed:
iOS 26.1，QMUINavigationTitleView可能会造成卡死的问题

CloudlessMoon committed:
修复 _QMUITransitionNavigationBar的bug

CloudlessMoon commented:
> @CloudlessMoon [85007d0](https://github.com/Tencent/QMUI_iOS/commit/85007d064d5e5f250c4982773832b546d809c383) // customBar.shouldPreventAppearance = YES; 你的这个修改，会出现 #1487 这个问题

你现在还能复现吗，复现环境说一下

CloudlessMoon committed:
qmui_isControllerRootView 设置为readonly

CloudlessMoon committed:
qmui_tintColorCustomized 应该在call super之前设置，否则tintColorDidChange拿不到当前值

CloudlessMoon committed:
_qmui_visible，处理UIWindow的情况

CloudlessMoon committed:
QMUITheme，UITextField也使用setNeedsDisplay，避免不必要的setAttributedText

CloudlessMoon committed:
QMUITheme，UITextView/UITextField对于回退到TextKit 1的输入框也执行 setNeedsDisplay

CloudlessMoon committed:
UISearchBar，修复qmui_placeholderColor、qmui_font 从后台切换到前台或者其他时机会被重置的问题

CloudlessMoon committed:
UISearchBar，cancelButton相关的代码回退到之前的逻辑，clearButton与cancelButton不是同一个概念

CloudlessMoon committed:
UIBarItem (QMUIBadge)，暂不支持qmui_badgeView和qmui_updatesIndicatorView，这里做一下标记

CloudlessMoon committed:
使用 performSelector
